ksql-engine/src/main/java/io/confluent/ksql/util/EngineProcessingLogMessageFactory.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License; you may not use this file
+ * except in compliance with the License.  You may obtain a copy of the License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.util;
+
+import io.confluent.ksql.GenericRow;
+import io.confluent.ksql.json.JsonMapper;
+import io.confluent.ksql.processing.log.ProcessingLogMessageSchema;
+import io.confluent.ksql.processing.log.ProcessingLogMessageSchema.MessageType;
+import java.util.function.Supplier;
+import org.apache.kafka.connect.data.SchemaAndValue;
+import org.apache.kafka.connect.data.Struct;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class EngineProcessingLogMessageFactory {
+  private static final Logger LOGGER
+      = LoggerFactory.getLogger(EngineProcessingLogMessageFactory.class);
+
+  private EngineProcessingLogMessageFactory() {
+  }
+
+  public static Supplier<SchemaAndValue> recordProcessingError(
+      final String errorMsg,
+      final GenericRow record
+  ) {
+    return () -> {
+      final Struct struct = new Struct(ProcessingLogMessageSchema.PROCESSING_LOG_SCHEMA);
+      struct.put(ProcessingLogMessageSchema.TYPE, MessageType.RECORD_PROCESSING_ERROR.ordinal());
```

Comment:
Can we avoid using `ordinal()`? There's no guarantees that this will be backwards compatible if new enum values are added. if we need to map an enum to a number, we can define it in the constructor for the enum.

See javadoc:
```
Most programmers will have no use for this method.  It is
designed for use by sophisticated enum-based data structures, such
as {@link java.util.EnumSet} and {@link java.util.EnumMap}.
```

Reply:
Good catch - I did not know this.